Crash on `:low`

**Version**: 30.r84.g741dd48
**OS**: Arch Linux x86-64
**Installation method**: with pacman, package extra/lf
**To reproduce**: open LF, navigate to a dir where the number of files/subdirs exceeds the number of terminal window rows, type `:low` and press return.
**Crash log**:
```
panic: runtime error: index out of range [-50]

goroutine 1 [running]:
main.(*nav).currFile(0x6061af?)
	github.com/horriblename/lf/nav.go:1993 +0x78
main.(*ui).loadFile(0xc000466820, 0xc0004c0000, 0x1)
	github.com/horriblename/lf/ui.go:700 +0x35
main.(*callExpr).eval(0xc00007abd0, 0xc0004c0000, {0xc00051d9d8?, 0x3?, 0x4?})
	github.com/horriblename/lf/eval.go:1529 +0x5e5
main.(*callExpr).eval(0xc00007b710, 0xc0004c0000, {0xc0000284e0?, 0x0?, 0xc0000061a0?})
	github.com/horriblename/lf/eval.go:2179 +0x8cf3
main.(*app).loop(0xc0004c0000)
	github.com/horriblename/lf/app.go:446 +0xd04
main.run()
	github.com/horriblename/lf/client.go:61 +0x3cf
main.main()
	github.com/horriblename/lf/main.go:331 +0x79e

```

Happens on 31.r16.g9698ecd from AUR as well.

--- a/nav.go
+++ b/nav.go
@@ -1144,7 +1144,7 @@ func (nav *nav) high() bool {
 
 	old := dir.ind
 	beg := max(dir.ind-dir.pos, 0)
-	offs := gOpts.scrolloff
+	offs := min(nav.height/2, gOpts.scrolloff)
 	if beg == 0 {
 		offs = 0
 	}
@@ -1175,7 +1175,14 @@ func (nav *nav) low() bool {
 	old := dir.ind
 	beg := max(dir.ind-dir.pos, 0)
 	end := min(beg+nav.height, len(dir.files))
-	offs := gOpts.scrolloff
+
+	offs := min(nav.height/2, gOpts.scrolloff)
+	// use a smaller value for half when the height is even and scrolloff is
+	// maxed in order to stay at the same row when using both high and low
+	if nav.height%2 == 0 {
+		offs = min(nav.height/2-1, gOpts.scrolloff)
+	}
+
 	if end == len(dir.files) {
 		offs = 0
 	}

